.github/workflows/reusable-release.yaml

```diff
@@ -0,0 +1,158 @@
+name: Reusable workflow that releases a flavor
+
+on:
+  workflow_call:
+    inputs:
+      base_image:
+        required: true
+        type: string
+      model:
+        required: false
+        type: string
+        default: "generic"
+      variant:
+        required: false
+        type: string
+        default: "core"
+      arch:
+        required: false
+        type: string
+        default: "amd64"
+      kubernets_version:
+        required: false
+        type: string
+      trusted_boot:
+        required: false
+        type: string
+        default: "false"
+
+jobs:
+  release:
+    runs-on: ${{ inputs.arch == 'arm64' && 'ubuntu-24.04-arm' || 'ubuntu-24.04' }}
+    name: ${{ inputs.base_image }}
+    permissions:
+      id-token: write
+      contents: write
+      actions: read
+      security-events: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Split base image
+        id: split
+        run: |
+          echo "flavor=$(echo ${{ inputs.base_image }} | cut -d ':' -f 1 | sed  's/\//-/g')" >> $GITHUB_OUTPUT
+          echo "flavor_release=$(echo ${{ inputs.base_image }} | cut -d ':' -f 2)" >> $GITHUB_OUTPUT
+      - name: Set Image Tag
+        run: |
+          echo "GIT_VERSION=$(git describe --always --tags --dirty)" >> $GITHUB_ENV
+          echo "IMAGE_TAG=quay.io/kairos/${{ steps.split.outputs.flavor }}:${{ steps.split.outputs.flavor_release }}-${{ inputs.variant }}-${{ inputs.arch }}-${{ inputs.model }}-${{ env.GIT_VERSION }}${{ inputs.kubernetes_version && '-'||''}}${{ inputs.kubernetes_version || ''}}${{ inputs.trusted_boot != 'false' && '-uki' || '' }}" >> $GITHUB_ENV
```

Comment:
Are we deprecating versioneer too? Because it's still used in `kairos-agent upgrade list-releases` and was also used when building the images, in order to prepare the values for `/etc/kairos-release`. If we still use it there, it would be good if the same tool was used everywhere to make sure we produce consistent names.

Another option would be to first build the image, then extract the artifact name from `/etc/kairos-release` in another step (from within the image) and finally push it with that name. But this creates a slightly more complicated workflow.

Reply:
oh I just saw, we already push in a second step so maybe the second option is not that more complicated.

Reply:
no, ideally we would use it although I did not see how to plug it in here properly so I went with the brute force approach of manually constructing it.

Ideally the versioneer should plug it in here easily, I just didnt have that in mind when dropping stuff xD

Reply:
Even if we use the same tool (versioneer), calling it twice could lead to inconsistencies so the best option imo is to just get the name from `/etc/kairos-release` in the generated artifact. Let's do that after merging this PR.